src/main/java/net/starlark/java/eval/StarlarkInt.java

```diff
@@ -599,11 +599,19 @@ public static StarlarkInt shiftLeft(StarlarkInt x, StarlarkInt y) throws EvalExc
   /** Returns x ^ y. */
   public static StarlarkInt xor(StarlarkInt x, StarlarkInt y) {
     if (x instanceof Int32 && y instanceof Int32) {
-      long xl = ((Int32) x).v;
-      long yl = ((Int32) y).v;
+      int xl = ((Int32) x).v;
+      int yl = ((Int32) y).v;
       return StarlarkInt.of(xl ^ yl);
     }
 
+    try {
```

Comment:
Is there any need to keep the int32 specialization if we have the int64 one? After all, the ALU is 64 bits wide. Doing so replaces two virtual calls with cheaper instanceof tests in the common case, but makes the uncommon case slower. Any idea whether it's a net win?

Reply:
Counter-intuitively, after removing special case, even Int32 test becomes faster. With this diff on top of this PR:

```
diff --git a/src/main/java/net/starlark/java/eval/StarlarkInt.java b/src/main/java/net/starlark/java/eval/StarlarkInt.java
index f588308966..6712c9b6e2 100644
--- a/src/main/java/net/starlark/java/eval/StarlarkInt.java
+++ b/src/main/java/net/starlark/java/eval/StarlarkInt.java
@@ -642,12 +642,6 @@ public abstract class StarlarkInt implements StarlarkValue, Comparable<StarlarkI

   /** Returns x & y. */
   public static StarlarkInt and(StarlarkInt x, StarlarkInt y) {
-    if (x instanceof Int32 && y instanceof Int32) {
-      int xl = ((Int32) x).v;
-      int yl = ((Int32) y).v;
-      return StarlarkInt.of(xl & yl);
-    }
-
     try {
       long xl = x.toLongFast();
       long yl = y.toLongFast();
```

For this benchmark:

```
def test():
    # ensure Int64 class is loaded by JVM
    ml = (1<<63) - 1 # maxlong

    for x in range(50000000):
        x = 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0

test()
```

It is faster:

```
A: n=26 mean=8.560 std=0.327 se=0.065 min=7.984 med=8.538
B: n=26 mean=8.293 std=0.278 se=0.055 min=7.880 med=8.333
B/A: 0.969 0.949..0.989 (95% conf)
```

I cannot clearly explain it, and results will likely be different on different JVM and different CPU, but I'll remove special cases for `Int32`.

```
$ java -version
openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)
$ sysctl -a | grep machdep.cpu.brand_string
machdep.cpu.brand_string: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
```